admin/services/data-access/pupil-restart.data.service.js

```diff
@@ -115,20 +116,79 @@ pupilRestartDataService.sqlFindLatestRestart = async function (pupilId) {
 
 /**
  * Find pupil's restart codes
- * @param pupilId
  * @return {Promise.<void>}
  */
 pupilRestartDataService.sqlFindRestartCodes = async function () {
   const sql = `
   SELECT 
     id, 
     code, 
-    statusDesc 
+    description
   FROM ${sqlService.adminSchema}.[pupilRestartCode]
-  ORDER BY statusDesc ASC`
+  ORDER BY description ASC`
+  return sqlService.query(sql)
+}
+
+/**
+ * Find pupil's restart reason description by id
+ * @param id
+ * @return {Promise.<void>}
+ */
+pupilRestartDataService.sqlFindRestartReasonDescById = async function (id) {
+  const sql = `
+  SELECT 
+    description
+  FROM ${sqlService.adminSchema}.[pupilRestartReason]
+  WHERE id=@id
+  ORDER BY description ASC`
+  const params = [
+    {
+      name: 'id',
+      value: id,
+      type: TYPES.Int
+    }
+  ]
+  const result = await sqlService.query(sql, params)
+  const obj = R.head(result)
+  return obj && obj['description']
+}
+
+/**
+ * Find restart reasons
+ * @return {Promise.<void>}
+ */
+pupilRestartDataService.sqlFindRestartReasons = async function () {
+  const sql = `
+  SELECT 
+    id, 
+    code, 
+    description
+  FROM ${sqlService.adminSchema}.[pupilRestartReason]
+  ORDER BY id ASC`
```

Comment:
Ordering by `id` seems odd. If the GUI wants the data to be ordered then add in a `displayOrder` field and order by that.  Ordering by `id` is a bad practise.

Reply:
@jon-shipley i need to sort the reason based on the way they have to appear in the front-end. if i don't order by id they will appear at a random order.
I am guessing the only way to solve this is to add an order field?

Reply:
I'd call it `displayOrder` or `guiOrder` to show that it is a display related field.  Please don't call it `order` as that is a reserved word (and `attendanceCode.order` is now tech debt.

Reply:
@jon-shipley updated using `displayOrder`